cli/src/db.ts

```diff
@@ -149,47 +49,166 @@ interface BulkRemovalResult {
   error?: Error
 }
 
-// TODO: test, then use this instead of single removals in position management
-export async function bulkRemovePositions(
-  address: Address,
-  positions: Position[]
-): Promise<BulkRemovalResult> {
-  const results: Record<string, RemovalResult> = {}
-  try {
-    const ids = positions.map((p) => buildId(address, p)).map((e) => e.id)
-    const r1 = await db.allDocs<Position>({ keys: ids })
-
-    const docs = r1.rows
-      .map((row) => {
-        const re = row as { key: string; error?: string }
-        if (re.error) {
-          results[re.key] = { id: re.key, removed: false, error: re.error }
-          return undefined
-        }
-        const rr = row as { id: string; value: { rev: string } }
-        return { _id: rr.id, _rev: rr.value.rev, _deleted: true }
+const useDb = () => {
```

Comment:
Why is this made into a hook?

Reply:
All options are passed into `Mainframe` as props, and I made `useDb` to get `db` option using hook

Reply:
Each time the hook is used, a new instance of the database is created. When there are two instances, they would conflict with each other and all except one instance would be locked out (since lock is a file). This would create problems unless the hook is used exactly in a single place, in which case it would make more sense to make it a shared context. But even as a shared context, there are risks of somehow having two instances, for example, when React re-render the whole mainframe due to recovery from error. This is why I initially put db strictly outside React context

Reply:
Yes, you are right. Didn't consider. Will update.